sidecar/sidecar.go

```diff
@@ -172,11 +150,7 @@ func Run(conf config.Config) error {
 		}
 	}
 
-	if conf.Supervise {
-		appSupervisor.DoAppSupervision(conf.App, registrationAgent)
-	} else {
-		select {}
-	}
+	appSupervisor.DoAppSupervision(registrationAgent)
```

Comment:
Why did you remove this? If there was no supervision, then you wouldn't need one right?
Specifically, how about this:
```
if conf.Supervise {
        appSupervisor := supervisor.NewAppSupervisor(&conf)
	appSupervisor.DoAppSupervision(conf.App, registrationAgent)
} else {
	select {}
}
```